### ISSUE TITLE
Questions about dstPort and srcPort from routing

### ISSUE THREAD

**Issue by terryops** [DaniilSokolyuk/go-pcap2socks#6]
Hi.
What are these two and how should I set it?
And it keeps prompting "2024/04/17 00:11:13 WARN [UDP] dial error:  error="proxy  not found"", why?
My config file is as below:

> 
> {
>   "pcap": {
>     "network": "10.10.0.0/24",
>     "localIP": "10.10.0.1"
>   },
>   "routing": {
>     "rules": [
>       {
>         "srcPort": "80,443,53,3544,1985,55960,4500,500,3075,1900,5355,5353",
>         "dstPort": "3074",
>         "outboundTag": "sout"
>       },
>       {
>         "srcIP": [
>           "10.10.0.0/24"
>         ],
>         "dstIP": [
>           "0.0.0.0/0"
>         
>         ]
>       }
>     ]
>   },
>   "outbounds": [
>    
>     {
>       "tag": "sout",
>       "socks": {
>         "address": "127.0.0.1:50325"
>       }
>     }
>   ]
> }

**Comment by DaniilSokolyuk** [comment]
hi!
your rule for srcIP and dstIP contains doesnt contains field [outboundTag](https://github.com/DaniilSokolyuk/go-pcap2socks/blob/main/config.json.fullsample#L27), by default its empty ("")
you specified dstIp 0.0.0.0.0/0 and so I don't think you need these rules, you can remove it
Also you should specified "" outbound tag for fallback, if srcPort or dstPort not be matched like [here](https://github.com/DaniilSokolyuk/go-pcap2socks/blob/main/config.json#L16)

**Comment by terryops** [comment]
I  got it running but failed after connected with a Nintendo switch(on screen is shows successfully connected to Internet.

config file:
```
{
  "pcap": {
    "network": "10.10.0.0/24",
    "localIP": "10.10.0.1"
  },
  "routing": {
    "rules": [
      {
        "dstPort": "80,443,53,3544,1985,55960,4500,500,3075,1900,5355,5353",
        "outboundTag": "sout"
      }
    ]
  },
  "outbounds": [
   
    {
      "tag": "sout",
      "socks": {
        "address": "127.0.0.1:53780",
        "username": "",
        "password": ""

      }
    },
     {
      "tag": "dir",
      "direct": {}
    }
  ]
}
```

below are the logs:
> 2024/04/17 14:06:15 INFO Enter this settings in your device's network settings ip=10.10.0.0/24 mask=255.255.255.0 gateway=10.10.0.1
w2024/04/17 14:06:19 INFO Device 10.10.0.4 (bc:ce:25:1d:05:c0) joined the network
2024/04/17 14:06:21 INFO [UDP] Connection source=10.10.0.4:52029 dest=8.8.8.8:53
2024/04/17 14:06:21 INFO [TCP] Connection source=10.10.0.4:52612 dest=23.204.139.148:80
2024/04/17 14:06:21 INFO [UDP] Connection source=10.10.0.4:52903 dest=8.8.8.8:53
2024/04/17 14:06:21 INFO [UDP] Connection source=10.10.0.4:21428 dest=8.8.8.8:53
2024/04/17 14:06:21 INFO [UDP] Connection source=10.10.0.4:27099 dest=8.8.8.8:53
2024/04/17 14:06:21 INFO [TCP] Connection source=10.10.0.4:52613 dest=23.51.137.35:443
2024/04/17 14:06:21 INFO [TCP] Connection source=10.10.0.4:52614 dest=23.40.148.176:443
2024/04/17 14:06:21 INFO [TCP] Connection source=10.10.0.4:52615 dest=107.21.233.83:443
2024/04/17 14:06:22 INFO [UDP] Connection source=10.10.0.4:54965 dest=8.8.8.8:53
2024/04/17 14:06:22 INFO [TCP] Connection source=10.10.0.4:52616 dest=52.70.41.32:443
2024/04/17 14:06:23 INFO [UDP] Connection source=10.10.0.4:52827 dest=8.8.8.8:53
2024/04/17 14:06:23 INFO [UDP] Connection source=10.10.0.4:22530 dest=8.8.8.8:53
2024/04/17 14:06:23 INFO [UDP] Connection source=10.10.0.4:40725 dest=8.8.8.8:53
2024/04/17 14:06:23 INFO [UDP] Connection source=10.10.0.4:22385 dest=8.8.8.8:53
2024/04/17 14:06:23 INFO [UDP] Connection source=10.10.0.4:42944 dest=8.8.8.8:53
2024/04/17 14:06:23 INFO [UDP] Connection source=10.10.0.4:12559 dest=8.8.8.8:53
2024/04/17 14:06:23 INFO [TCP] Connection source=10.10.0.4:52617 dest=3.223.147.81:443
2024/04/17 14:06:23 INFO [TCP] Connection source=10.10.0.4:52618 dest=184.27.185.9:443
2024/04/17 14:06:23 INFO [TCP] Connection source=10.10.0.4:52619 dest=23.46.63.155:443
2024/04/17 14:06:23 INFO [TCP] Connection source=10.10.0.4:52620 dest=23.40.148.176:443
2024/04/17 14:06:23 INFO [TCP] Connection source=10.10.0.4:52621 dest=34.232.137.186:443
2024/04/17 14:06:23 INFO [TCP] Connection source=10.10.0.4:52622 dest=34.238.5.172:443
2024/04/17 14:06:24 INFO [UDP] Connection source=10.10.0.4:19832 dest=8.8.8.8:53
2024/04/17 14:06:24 INFO [TCP] Connection source=10.10.0.4:52623 dest=3.232.109.210:443
2024/04/17 14:06:25 INFO [UDP] Connection source=10.10.0.4:62609 dest=8.8.8.8:53
2024/04/17 14:06:25 INFO [TCP] Connection source=10.10.0.4:52624 dest=18.205.95.65:443
2024/04/17 14:06:25 INFO [UDP] Connection source=10.10.0.4:48444 dest=8.8.8.8:53
2024/04/17 14:06:26 INFO [TCP] Connection source=10.10.0.4:52625 dest=18.65.216.65:443
2024/04/17 14:06:26 INFO [UDP] Connection source=10.10.0.4:13330 dest=8.8.8.8:53
2024/04/17 14:06:26 INFO [UDP] Connection source=10.10.0.4:17545 dest=8.8.8.8:53
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x28 pc=0x10519c138]

goroutine 267 [running]:
github.com/DaniilSokolyuk/go-pcap2socks/proxy.(*Router).DialUDP(0x140002594d0, 0x14000132900)
	/Users/terrygu/Downloads/go-pcap2socks-main/proxy/router.go:55 +0x158
github.com/DaniilSokolyuk/go-pcap2socks/proxy.DialUDP(...)
	/Users/terrygu/Downloads/go-pcap2socks-main/proxy/proxy.go:48
github.com/DaniilSokolyuk/go-pcap2socks/tunnel.HandleUDPConn({0x1054497f0, 0x14000333380})
	/Users/terrygu/Downloads/go-pcap2socks-main/tunnel/udp.go:21 +0x6c
github.com/DaniilSokolyuk/go-pcap2socks/core.Tunnel.HandleUDP(...)
	/Users/terrygu/Downloads/go-pcap2socks-main/core/tunnel.go:17
github.com/DaniilSokolyuk/go-pcap2socks/core.handler.NewPacketConnection(...)
	/Users/terrygu/Downloads/go-pcap2socks-main/core/udpforwarder.go:44
github.com/sagernet/sing/common/udpnat.(*Service[...]).NewContextPacket.func2()
	/Users/Terrygu/go/pkg/mod/github.com/sagernet/sing@v0.3.8/common/udpnat/service.go:82 +0x98
created by github.com/sagernet/sing/common/udpnat.(*Service[...]).NewContextPacket in goroutine 12
	/Users/Terrygu/go/pkg/mod/github.com/sagernet/sing@v0.3.8/common/udpnat/service.go:81 +0x2b4

**Comment by DaniilSokolyuk** [comment]
As I said earlier, it's because there is no default outbound tag (tag = "")
if you need proxy all traffics to the socks5 use this
```
{
  "pcap": {
    "network": "10.10.0.0/24",
    "localIP": "10.10.0.1"
  },
  "routing": {
    "rules": []
  },
  "outbounds": [
    {
      "tag": "",
      "socks": {
        "address": "127.0.0.1:50325"
      }
    }
  ]
}
```

**Comment by terryops** [comment]
Thanks, it works much better now. But sometimes I still got 
Error like:
2024/04/18 11:32:15 ERROR udp read packet error:  !BADKEY="io: read/write on closed pipe"
2024/04/18 11:32:15 ERROR udp read packet error:  !BADKEY="io: read/write on closed pipe"
2024/04/18 11:32:15 WARN [UDP] pipe closed source=10.10.0.4:65391 dest=[::]:57376 error="io: read/write on closed pipe"

Is there anything I can do? Is it the problem of the proxy?

**Comment by DaniilSokolyuk** [comment]
its ok, UDP does not establish a connection when sending data, its a just timeouts on port listen, not a problem
